spec/tbd_tests_spec.rb

```diff
@@ -1582,6 +1582,177 @@
+    expect(os_model2.empty?).to be(false)
+    os_model2 = os_model2.get
+
+    io_path2 = File.dirname(__FILE__) + "/../json/tbd_warehouse.out.json"
+    io2, surfaces = processTBD(os_model2, psi_set, io_path2, schema_path)
+    expect(surfaces.size).to eq(23)
+
+    surfaces.each do |id, surface|
+      next unless surface.has_key?(:edges)
+      os_model.getSurfaces.each do |s|
+        next unless id == s.nameString
+        expect(s.isConstructionDefaulted).to be(false)
+        expect(/ tbd/i.match(s.construction.get.nameString)).to_not eq(nil)
+      end
+    end
+
+    surfaces.each do |id, surface|
+      if surface.has_key?(:ratio)
+        ratio  = format "%3.1f", surface[:ratio]
+        name   = id.rjust(15, " ")
+        # puts "#{name} RSi derated by #{ratio}%"
+        # ... should print out:
+        # Fine Storage Office Front Wall  RSi derated by - 8.4%
+        # Office Left Wall                RSi derated by -36.8% ~+5% increase
+        # Fine Storage Office Left Wall   RSi derated by - 9.8%
+        # Fine Storage Roof               RSi derated by - 4.4%
+        # Office Front Wall               RSi derated by -22.6%
+        # Fine Storage Right Wall         RSi derated by - 8.8%
+        # Bulk Storage Right Wall         RSi derated by - 8.7%
+        # Bulk Storage Rear Wall          RSi derated by - 8.3%
+        # Fine Storage Front Wall         RSi derated by - 8.5%
+        # Fine Storage Left Wall          RSi derated by - 8.4%
+        # Bulk Storage Left Wall          RSi derated by -11.8%
+        # Bulk Storage Roof               RSi derated by - 4.0%
+        next unless name == "Office Left Wall"
+        expect(surface[:ratio]).to be_within(0.2).of(-36.8)
+
+        # Note the content of the out.json file:
+        # {
+        #  "psi": "bad",
+        #  "type": "fenestration",
+        #  "length": 1.524940613884627,
+        #  "surfaces": [
+        #    "Office Left Wall Window1",
+        #    "Office Left Wall"
+        #  ],
+        #  "count": 2
+        # },
+        # {
+        #  "psi": "bad",
+        #  "type": "fenestration",
+        #  "length": 2.13349595113501,
+        #  "surfaces": [
+        #    "Office Left Wall Window1",
+        #    "Office Left Wall"
+        #  ],
+        #  "count": 2
+        # },
+        #
+        # 2x entries (instead of 1x):
+        #   - 2x 1.52m edges
+        #   - 2x 2.13m edges
+        # ... otherwise, same PSI set, PSI type and surface list
+      else
+        expect(surface[:boundary].downcase).to_not eq("outdoors")
+      end
+    end
+
+    # Now mimic (again) the export functionality of the measure
+    out2 = JSON.pretty_generate(io2)
+    out_path2 = File.dirname(__FILE__) + "/../json/tbd_warehouse2.out.json"
+    File.open(out_path2, "w") do |out_path2|
+      out_path2.puts out2
+    end
+
+    # Both output files should be the same ...
+    cmd = "diff #{out_path} #{out_path2}"
+    expect(system( cmd )).to be(true)
+  end
```

Comment:
@macumber : Is it considered legit to run system commands as tests?

Reply:
Yes, only issue here is that `diff` might not be present on some Windows machine (I have it because I have Git Bash)

Reply:
Pure Ruby way could be to read both files in as strings and do a string equality check

Reply:
OK. Good points. Will look into this (follow up commit).